apps/microsoft-teams/frontend/src/hooks/useGetTeamsChannels.ts

```diff
@@ -12,24 +14,42 @@ const useGetTeamsChannels = () => {
   const getAllChannels = useCallback(async () => {
     try {
       setLoading(true);
-      const { response } = await sdk.cma.appActionCall.createWithResponse(
+      const { sys: appActionCallSys } = await sdk.cma.appActionCall.createWithResult(
         {
           appActionId: 'msteamsListChannels',
           environmentId: sdk.ids.environment,
           spaceId: sdk.ids.space,
           appDefinitionId: sdk.ids.app!,
-          userId: sdk.ids.user,
         },
         {
           parameters: {},
         }
       );
-      const body = JSON.parse(response.body);
-      if (body.ok) {
-        setChannels(body.data);
-        setError(undefined);
+
+      if (appActionCallSys.status === 'succeeded') {
+        const { result: appActionResult } = appActionCallSys;
+        assertAppActionResult<TeamsChannel[]>(appActionResult);
+
+        if (appActionResult.ok) {
+          setChannels((appActionResult as AppActionResultSuccess<TeamsChannel[]>).data || []);
```

Comment:
Hm, this type assertion should not be needed? TS should already have the correct type via the assertion above.